src/lualib/Promise.ts

```diff
@@ -0,0 +1,335 @@
+declare function error( value:string, level?:number ): never;
+const logError: any = print;
+
+let runAtNextTick: ( func: () => void ) => void = func => func();
+function setRunAtNextTickFunction(  func: (f: () => void ) => void ): void {
```

Comment:
IMO it would be better to:
a) `Promise.runAtNextTick` (overridable)
b) `Promise.setRunAtNextTickFunction`
c) `Promise._setRunAtNextTickFunction`
d) Introduce some namespace for non-standard stuff, like `tstl.setRunAtNextTickFunction`

Reply:
I chose option c). To set the tick-function you can no use something like this in your *.ts files:
```typescript
interface Promise {
   _setRunAtNextTickFunction( func: ( func: () => void ) => void ): void;
}
Promise._setRunAtNextTickFunction( ( func ) => func() );
```